### PR TITLE
refactor(superset-ui-core): Migrate FallbackComponent.test to RTL

### PR DESCRIPTION
**Pull request by rtexelm**
### SUMMARY

Simple changes to utilize RTL instead of enzyme

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/test/chart/components/FallbackComponent.test.tsx [hunk 1]
```diff
@@ -18,42 +18,49 @@
...
-    const messages = wrapper.find('code');
-    expect(messages).toHaveLength(2);
-    expect(messages.at(0).text()).toEqual('Error: CaffeineOverLoadException');
-    expect(messages.at(1).text()).toEqual('Error at line 1: x.drink(coffee)');
-  });
+test('renders error and stack trace', () => {
+  const { getByText } = render(
+    <ThemeProvider theme={supersetTheme}>
+      <FallbackComponent componentStack={STACK_TRACE} error={ERROR} />
+    </ThemeProvider>,
```
Comment (eschutho) [inline superset-frontend/packages/superset-ui-core/test/chart/components/FallbackComponent.test.tsx:34]: just a nit and not a blocker, but you can dry up this render function here if you want.

**Review by eschutho — APPROVED** [review]
LGTM, one small nit (not blocking)

**Comment by rtexelm** [comment]
@eschutho thanks for reminding me, forgot in my haste to get a PR up. this is what you're thinking of, right?

**Comment by eschutho** [comment]
@rtexelm yeah, that's perfect!

**Review by eschutho — APPROVED** [review]
thanks @rtexelm!

**Pull request merged**